status/conditions_test.go

```diff
@@ -69,198 +76,389 @@ func withLastTransitionTime(c Condition, t time.Time) Condition {
-		generateCondition("True", corev1.ConditionTrue),
-		generateCondition("Unknown", corev1.ConditionUnknown),
-	)
-
-	assert.False(t, conditions.IsFalseFor(ConditionType("True")))
-	assert.True(t, conditions.IsFalseFor(ConditionType("False")))
-	assert.False(t, conditions.IsFalseFor(ConditionType("Unknown")))
-	assert.False(t, conditions.IsFalseFor(ConditionType("DoesNotExist")))
-}
+			Describe("Testing if False for conditiontype False", func() {
+				It(" should be False for IsTrueFor", func() {
+					Expect(conditions.IsTrueFor(ConditionType("False"))).Should(BeFalse())
+				})
+			})
+
+			Describe("Testing if False for conditiontype Unknown", func() {
+				It(" should be False for IsTrueFor", func() {
+					Expect(conditions.IsTrueFor(ConditionType("Unknown"))).Should(BeFalse())
+				})
+			})
+
+			Describe("Testing if False for conditiontype DoesNotExist", func() {
+				It(" should be False for IsTrueFor", func() {
+					Expect(conditions.IsTrueFor(ConditionType("DoesNotExist"))).Should(BeFalse())
+				})
+			})
+
+		})
+
+		Describe("TestConditionsIsFalseFor", func() {
+
+			Describe("Testing if False for conditiontype True", func() {
+				It(" should be False for IsTrueFor", func() {
+					Expect(conditions.IsFalseFor(ConditionType("True"))).Should(BeFalse())
+				})
+			})
+
+			Describe("Testing if True for conditiontype False", func() {
+				It(" should be True for IsTrueFor", func() {
+					Expect(conditions.IsFalseFor(ConditionType("False"))).Should(BeTrue())
+				})
+			})
+
+			Describe("Testing if False for conditiontype Unknown", func() {
+				It(" should be False for IsTrueFor", func() {
+					Expect(conditions.IsFalseFor(ConditionType("Unknown"))).Should(BeFalse())
+				})
+			})
+
+			Describe("Testing if False for conditiontype DoesNotExist", func() {
+				It(" should be False for IsTrueFor", func() {
+					Expect(conditions.IsFalseFor(ConditionType("DoesNotExist"))).Should(BeFalse())
+				})
+			})
+
+		})
+
+		Describe("TestConditionsIsUnknownFor", func() {
+
+			Describe("Testing if False for conditiontype True", func() {
+				It(" should be False for IsUnknownFor", func() {
+					Expect(conditions.IsUnknownFor(ConditionType("True"))).Should(BeFalse())
+				})
+			})
+
+			Describe("Testing if True for conditiontype False", func() {
+				It(" should be False for IsUnknownFor", func() {
+					Expect(conditions.IsUnknownFor(ConditionType("False"))).Should(BeFalse())
+				})
+			})
+
+			Describe("Testing if True for conditiontype Unknown", func() {
+				It(" should be True for IsUnknownFor", func() {
+					Expect(conditions.IsUnknownFor(ConditionType("Unknown"))).Should(BeTrue())
+				})
+			})
+
+			Describe("Testing if True for conditiontype DoesNotExist", func() {
+				It(" should be True for IsUnknownFor", func() {
+					Expect(conditions.IsUnknownFor(ConditionType("DoesNotExist"))).Should(BeTrue())
+				})
+			})
+
+		})
+
+	})
+
+	Describe("TestConditionsMarshalUnmarshalJSON", func() {
```

Comment:
I would do a similar thing as I stated above with `Conditions`. 
```golang
var _ = Describe("Conditions", func() {
    Describe("NewConditions", func() {
        // ...
    }
    Describe("IsTrueFor", func() {
        // ...
    }
    Describe("IsFalseFor", func() {
        // ...
    }
    Describe("IsUnknownFor", func() {
        // ...
    }
    Describe("SetCondition", func() {
        // ...
    }
    Describe("GetCondition", func() {
        // ...
    }
    Describe("RemoveCondition", func() {
        // ...
    }
}
```
Then for each of the above `Describe`s you would have one or more `It` statements to test the positive and negative outputs.